src/main/java/com/zendesk/maxwell/MaxwellParser.java

```diff
@@ -117,143 +177,10 @@ private void processQueryEvent(QueryEvent event) throws SchemaSyncError {
-
-			Table tbl = db.findTable(tblName);
-			if ( tbl == null )
-				throw new RuntimeException("Couldn't find table " + tblName);
-
-			tableMapCache.put(tableId, tbl);
-		}
-	}
-
-	private Table getTableForId(Long tableId) {
-		return tableMapCache.get(tableId);
-	}
-
-	private BinlogEventV4 getBinlogEvent() throws Exception {
-		BinlogEventV4 event;
-
-		if (parser == null) {
-			initParser(fileName, startPosition);
-			parser.start();
-		}
-
-		while (true) {
-			event = queue.poll(100, TimeUnit.MILLISECONDS);
-			if (event != null) {
-				if ( event instanceof RotateEvent ) {
-					RotateEvent r = (RotateEvent) event;
-					// we throw out the old parser and boot a new one.
-					initParser(r.getBinlogFileName().toString(), r.getBinlogPosition());
-					this.parser.start();
-				} else {
-					return event;
-				}
-			} else {
-				if (!this.parser.isRunning()) {
-					/* parser stopped and the queue is empty.
-					   Likely that we've simply hit the end, but another possibility
-					   is that the server crashed and never inserted the "rotate" event in
-				       the logs.  Let's test for that. */
-
-					String nextFile = findNextBinlogFile(this.fileName);
-					if (nextFile == null)
-						return null;
-					else {
-						initParser(nextFile, 4);
-						this.parser.start();
-					}
-				}
-			}
-		}
-	}
-
-	private String findNextBinlogFile(String fileName) {
-		Pattern p = Pattern.compile("(.*)\\.(\\d+)");
-		Matcher m = p.matcher(fileName);
-		if ( !m.matches() )
-			return null;
-
-		Integer nextInt = Integer.valueOf(m.group(2)) + 1;
-		String testFile = m.group(1) + "." + String.format("%06d", nextInt);
-		File f = new File(this.filePath + "/" + testFile);
-		if ( f.exists() )
-			return testFile;
-		else
-			return null;
-	}
-
-	protected FileBasedBinlogParser getDefaultBinlogParser() throws Exception {
-		final FileBasedBinlogParser r = new FileBasedBinlogParser();
-		r.registgerEventParser(new StopEventParser());
-		r.registgerEventParser(new RotateEventParser());
-		r.registgerEventParser(new IntvarEventParser());
-		r.registgerEventParser(new XidEventParser());
-		r.registgerEventParser(new RandEventParser());
-		r.registgerEventParser(new QueryEventParser());
-		r.registgerEventParser(new UserVarEventParser());
-		r.registgerEventParser(new IncidentEventParser());
-		r.registgerEventParser(new TableMapEventParser());
-		r.registgerEventParser(new WriteRowsEventParser());
-		r.registgerEventParser(new UpdateRowsEventParser());
-		r.registgerEventParser(new DeleteRowsEventParser());
-		r.registgerEventParser(new WriteRowsEventV2Parser());
-		r.registgerEventParser(new UpdateRowsEventV2Parser());
-		r.registgerEventParser(new DeleteRowsEventV2Parser());
-		r.registgerEventParser(new FormatDescriptionEventParser());
-
-		return r;
-	}
-
-	private void initParser(String fileName, long position) throws Exception {
-		FileBasedBinlogParser bp = getDefaultBinlogParser();
-
-		bp.setStartPosition(position);
```

Comment:
we're using the OpenReplicator's objects directly instead of implementing our own.